pkg/catalog/show.go

```diff
@@ -4,20 +4,64 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"net/url"
+	"path"
 
 	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
 )
 
+// ShowOptions provides options to the show command.
+type ShowOptions struct {
+	catalogName    string
+	profileName    string
+	catalogVersion string
+}
+
+// ShowOpts is the functional options for Show
+type ShowOpts func(option *ShowOptions)
+
+// WithCatalogName sets up catalog name for Show
+func WithCatalogName(name string) ShowOpts {
+	return func(option *ShowOptions) {
+		option.catalogName = name
+	}
+}
+
+// WithProfileName sets up the profile name for Show
+func WithProfileName(name string) ShowOpts {
+	return func(option *ShowOptions) {
+		option.profileName = name
+	}
+}
+
+// WithCatalogVersion sets up catalog version for Show
+func WithCatalogVersion(version string) ShowOpts {
+	return func(option *ShowOptions) {
+		option.catalogVersion = version
+	}
+}
+
 // Show queries the catalog at catalogURL for a profile matching the provided profileName
-func Show(catalogClient CatalogClient, catalogName, profileName string) (profilesv1.ProfileDescription, error) {
-	data, code, err := catalogClient.DoRequest(fmt.Sprintf("/profiles/%s/%s", catalogName, profileName), nil)
+func Show(catalogClient CatalogClient, opts ...ShowOpts) (profilesv1.ProfileDescription, error) {
+	options := &ShowOptions{}
+	for _, o := range opts {
+		o(options)
+	}
```

Comment:
How come you decided on the `opts` pattern? I think its slightly confusing because `profileName` and `catalogName` are mandatory, which doesn't really get conveyed with this pattern

Reply:
funny that you ask. :D :D :D It's a nice story. So, I started with defining version and initially, only install cared about it, and it was strange adding it to show and then mostly I called it with `""` because show wasn't using it actually from anywhere else. But then I realised that now show can also use it, so I started defining it. And previously I didn't use url.Join in this nice way either. :D so I was checking if version is empty, which was ugly as well. :DDDD

ANYWAYS. Now I think it's obsolete too and was thinking about removing it, but wanted to know what you think about it. :D